include/sys/sys_heap.h

```diff
@@ -82,6 +83,17 @@ struct sys_heap_runtime_stats {
 int sys_heap_runtime_stats_get(struct sys_heap *heap,
 		struct sys_heap_runtime_stats *stats);
 
+/**
+ * @brief Reset the historical runtime statistics of a sys_heap
+ *
+ * Reset the sys_heap runtime statistics that aggregate historical data, such as
+ * the maximum number of allocated bytes, to the values that reflect the current
+ * state of affairs only.
+ *
+ * @param heap Pointer to sys_heap
+ * @return -EINVAL if null pointer was passed, otherwise 0
+ */
+int sys_heap_runtime_stats_reset_historical(struct sys_heap *heap);
```

Comment:
Maybe `_update_` instead of `_reset_`? Reset, for some, might mean setting it back to zero.

Reply:
Hmm. Isn't resetting more about setting back to the initial value rather than to zero? I thought that for the **maximum** kinds of statistics, the current value is a good initial value, but maybe it's just me. On the other hand, `_update_` may indicate many different things, so I would prefer to assign it a more descriptive name. What about something like `_remove_history` or `_history_restart`?

Reply:
hm... in this context, heap is initially empty so max allocated bytes would be zero. Maybe `_init_historical()`?

Reply:
ok, I have done the change.